### ISSUE TITLE
Shapely availability in QGIS

### ISSUE THREAD

**Issue by Dan-Eli** [qgis/QGIS-Documentation#5778]
I noticed that at least since QGIS 3.4 Shapely library is part of the regular QGIS distributions. I know it was not always the case in earlier distribution.

Is it now "normal" for QGIS to include Shapley in the regular distribution? I would suggest to include somewhere in the documentation a page listing the libraries included in the distribution and for each the version of the library.

Like this, it would be clearer for developper what is accessible from each QGISdistribution.

Note 1: Very good idea to include Shapely in QGIS distribution. I used a lot Shapely to develop script and it is now easier to port these script into QGIS Processing with Shapley part of the regular distribution.

Note2: This is true for Windows distribution. I do not know the status for Mac and Linux distribution

**Comment by DelazJ** [comment]
Hi @Dan-Eli 
Are you sure shapely is part of **QGIS** distribution? I can see python*-shapely in my osgeo4w packages but I don't have them installed and my QGIS is working without problem. Maybe it was not part of the **osgeo4w** (which is not the same thing if I may) packages earlier? 
About listing libraries, there's a thread in the QGIS-Developer mailing list that suggests that the dependencies versions may vary depending on the OS. I think that what you'll for sure find in any distribution of QGIS is [this (and derivatives)](https://github.com/qgis/QGIS/blob/master/INSTALL.md#2-overview). But I don't know if this is what you are looking for.

I'm not convinced this request is something for the documentation...

**Comment by Dan-Eli** [comment]
Hi @DelazJ 
I think it's part of it otherwise I do not know how the following lines could work:

In QGIS python console you can type:

```
import shapely
print (shapely.__file__)
```

The reply is

`C:\PROGRA~1\QGIS3~1.14\apps\Python37\lib\site-packages\shapely\__init__.py`

So, it's part of the QGIS distribution and installed

This is working on Windows at least with QGIS 3.4, 3.8 and 3.14

Why shapley is distributed?
Will it be part of future distribution?
Shapely is very usefull when writing processing script
I find no trace of Shapely in the documentation... but it's in the distribution (according to my test)

**Comment by ghtmtt** [comment]
Yes, shapely is distributed with QGIS with OSGEO4W. @jef-n is the maintainer of the packages and can point to the list of available libraries.

**Comment by DelazJ** [comment]
@Dan-Eli Here the result of your commands (I don't use standalone installs that said)
![image](https://user-images.githubusercontent.com/7983394/89954896-33937a00-dc32-11ea-8389-641539bdab83.png)
@ghtmtt yes, this is what i was saying above but the question is: is it installed with QGIS (at least with standalone install, because my "default" osgeo4w install doesn't seem to need it)?

**Comment by jef-n** [comment]
It's a dependency of qgis-full, which includes more optional packages.  The standalones are based on qgis-full.

**Comment by ghtmtt** [comment]
thanks for the clarification @jef-n. @DelazJ maybe it is worth to mention in the docs which packages are shipped with the different installations. What do you think?

**Comment by jef-n** [comment]
> @DelazJ maybe it is worth to mention in the docs which packages are shipped with the different installations. What do you think?

Just include pointers.  A list will quickly date out.

**Comment by DelazJ** [comment]
> Just include pointers. A list will quickly date out.

Agreed. @jef-n Any link to the information?

**Comment by jef-n** [comment]
> Agreed. @jef-n Any link to the information?

No, at least nothing that is easily digestible.  The pointer could be "Install qgis-full / qgis-ltr-full from OSGeo4W to get a good set of extra packages (eg. satisfying requirements of a lot of third-party plugins).  The standalone installer is also based on these sets."

**Comment by m-kuhn** [comment]
@DelazJ this may evolve over time (especially the version) is this something we want to maintain?
I don't have something readily available.

We could

- detect from an installation https://stackoverflow.com/a/740018/31852289
- or export at build time (and show in about?)
